File: opensrp-chw-referral/src/main/java/org/smartregister/chw/referral/domain/ReferralFollowupObject.java
package org.smartregister.chw.referral.domain;

import com.google.gson.annotations.SerializedName;

import org.smartregister.chw.referral.util.DBConstants;
import org.smartregister.commonregistry.CommonPersonObjectClient;

import java.io.Serializable;

public class ReferralFollowupObject implements Serializable {
    private String relationalId;
    private String details;
    private String baseEntityId;

    @SerializedName("chw_followup_feedback")
    private String chwFollowupFeedback;

    @SerializedName("chw_followup_feedback_id")
    private String chwFollowupFeedbackId;

    @SerializedName("other_followup_feedback_information")
    private String otherFollowupFeedbackInformation;

    @SerializedName("chw_followup_date")
    private String chwFollowupDate;
    private boolean isClosed;

    public ReferralFollowupObject() {
    }

    public ReferralFollowupObject(CommonPersonObjectClient client) {
        baseEntityId = client.getColumnmaps().get(DBConstants.Key.BASE_ENTITY_ID);
        relationalId = client.getColumnmaps().get(DBConstants.Key.RELATIONAL_ID);
        chwFollowupFeedbackId = client.getColumnmaps().get(DBConstants.Key.CHW_FOLLOWUP_FEEDBACK_ID);
        otherFollowupFeedbackInformation = client.getColumnmaps().get(DBConstants.Key.OTHER_FOLLOWUP_FEEDBACK_INFORMATION);
        chwFollowupDate = client.getColumnmaps().get(DBConstants.Key.CHW_FOLLOWUP_DATE);

        details = client.getColumnmaps().get(DBConstants.Key.DETAILS);
        isClosed = Boolean.parseBoolean(client.getColumnmaps().get(DBConstants.Key.IS_CLOSED));
    }

    public String getDetails() {
        return details;
    }

    public void setDetails(String details) {
        this.details = details;
    }

    public String getBaseEntityId() {
        return baseEntityId;
    }

    public void setBaseEntityId(String baseEntityId) {
        this.baseEntityId = baseEntityId;
    }

    public String getChwFollowupFeedback() {
        return chwFollowupFeedback;
    }

    public void setChwFollowupFeedback(String chwFollowupFeedback) {
        this.chwFollowupFeedback = chwFollowupFeedback;
    }

    public String getOtherFollowupFeedbackInformation() {
        return otherFollowupFeedbackInformation;
    }

    public void setOtherFollowupFeedbackInformation(String otherFollowupFeedbackInformation) {
        this.otherFollowupFeedbackInformation = otherFollowupFeedbackInformation;
    }

    public boolean getIsClosed() {
        return isClosed;
    }

    public void setIsClosed(boolean closed) {
        isClosed = closed;
    }

    public String getRelationalId() {
        return relationalId;
    }

    public void setRelationalId(String relationalid) {
        this.relationalId = relationalid;
    }

    public String getChwFollowupDate() {
        return chwFollowupDate;
    }

    public void setChwFollowupDate(String chwFollowupDate) {
        this.chwFollowupDate = chwFollowupDate;
    }

    public String getChwFollowupFeedbackId() {
        return chwFollowupFeedbackId;
    }

    public void setChwFollowupFeedbackId(String chwFollowupFeedbackId) {
        this.chwFollowupFeedbackId = chwFollowupFeedbackId;
    }
}

File: opensrp-chw-referral/src/main/java/org/smartregister/chw/referral/domain/ReferralFollowupObject.kt
package org.smartregister.chw.referral.domain

import com.google.gson.annotations.SerializedName
import org.smartregister.chw.referral.util.DBConstants
import org.smartregister.commonregistry.CommonPersonObjectClient
import java.io.Serializable

class ReferralFollowupObject : Serializable {

    var relationalId: String? = null
    var details: String? = null
    var baseEntityId: String? = null
    var isClosed = false

    @SerializedName("chw_followup_feedback")
    var chwFollowupFeedback: String? = null

    @SerializedName("chw_followup_feedback_id")
    var chwFollowupFeedbackId: String? = null

    @SerializedName("other_followup_feedback_information")
    var otherFollowupFeedbackInformation: String? = null

    @SerializedName("chw_followup_date")
    var chwFollowupDate: String? = null

    constructor()

    constructor(client: CommonPersonObjectClient) {
        baseEntityId = client.columnmaps[DBConstants.Key.BASE_ENTITY_ID]
        relationalId = client.columnmaps[DBConstants.Key.RELATIONAL_ID]
        chwFollowupFeedbackId = client.columnmaps[DBConstants.Key.CHW_FOLLOWUP_FEEDBACK_ID]
        otherFollowupFeedbackInformation =
            client.columnmaps[DBConstants.Key.OTHER_FOLLOWUP_FEEDBACK_INFORMATION]
        chwFollowupDate = client.columnmaps[DBConstants.Key.CHW_FOLLOWUP_DATE]
        details = client.columnmaps[DBConstants.Key.DETAILS]
        isClosed = client.columnmaps[DBConstants.Key.IS_CLOSED] == "1"
    }

}